amm/repl/src/test/scala/ammonite/session/ImportTests.scala

```diff
@@ -330,5 +330,25 @@ object ImportTests extends TestSuite{
         res3: Int = 1
       """)
     }
+    test("shapelessBugMinimized"){
+      check.session("""
+        @ object f{ case class Foo()}
+
+        @ import f.Foo
+
+        @ val Foo = 1
+
+        @ Foo
+      """)
+    }
+    test("shapelessBugFull"){
```

Comment:
I guess this test should be guarded by a `if (check.scala2)`, as shapeless basically [isn't the same project](https://github.com/typelevel/shapeless-3) in Scala 3.